x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/transport/SSLExceptionHelper.java

```diff
@@ -32,4 +33,10 @@ public static boolean isReceivedCertificateUnknownException(Throwable e) {
                 && e.getCause() instanceof SSLException
                 && "Received fatal alert: certificate_unknown".equals(e.getCause().getMessage());
     }
+
+    public static boolean isNoCertificateException(Throwable e) {
+        return e instanceof DecoderException
+                && e.getCause() instanceof SSLHandshakeException
+                && "null cert chain".equals(e.getCause().getMessage());
```

Comment:
I'm not sure this is safe:

1. This assumes that the message is the same across all supported JDKs here (and it seems it changed in JDK12)
2. Isn't this same message also thrown when there's simply no common cert chain with the client, so this might be a problem on either the server or client?

=> I don't have the experience with SSL to fully judge the second point, but the first point makes me not want to make this change. At the very least, if we want to make it, we should have a test for this so we can ensure it works the same across all the JVMs we test on?